test/state/stateV2_v3_migration.test.ts

```diff
@@ -0,0 +1,131 @@
+import { NetworkIdTypes } from "../../helpers/NetworkIdTypes";
+
+const stateTransitionsWithProofs = [
+    require("./data/user_state_genesis_transition.json"),
+    require("./data/user_state_next_transition.json"),
+  ];
+
+  const stateTransitionsWithNoProofs = [
+    {
+      id: '6901746346790563787434755862277025452451108972170386555162524223864832',
+      oldState: '1099511627776',
+      newState: '2199023255552',
+      isOldStateGenesis: true,
+    },
+    {
+      id: '6901746346790563787434755862277025452451108972170386555162524223864832',
+      oldState: '2199023255552',
+      newState: '3298534883328',
+      isOldStateGenesis: false,
+    },
+  ];
+
+describe.only("Deploy StateV2 v2 and migrate", () => {
+    let state;
+    let guWrpr;
+    let deployHelper;
+    let signers;
+    const defaultIdType = NetworkIdTypes.polygonMumbai;
+
+    before(async function () {
+        signers = await ethers.getSigners();
+        deployHelper = await DeployHelper.initialize();
+        const contracts = await deployHelper.deployStateV2('VerifierV2', 'StateV2_deployed', true);
+        state = contracts.state;
+        guWrpr = await deployHelper.deployGenesisUtilsWrapper();
+  });
+
+  it("Check migration", async () => {
+    // 1. init old contract by abi & address
+    const stateContractMigrationHelper = new StateContractMigrationHelper(deployHelper, signers[0]);
+    const oldContractABI = require("../../scripts/StateV2_deployed_abi.json");  // abi of contract that will be upgraded
+    const stateContractAddress = state.address;  // address of contract that will be upgraded
+    const stateContractInstance = await stateContractMigrationHelper.getInitContract({
+        contractNameOrAbi: oldContractABI,
+        address: stateContractAddress,
+    });
+
+    // 2. publish first state
+    const params1 = await publishState(stateContractInstance, stateTransitionsWithProofs[0]);
+    const res1 = await stateContractInstance.getStateInfoById(params1.id);
+    expect(res1.state).to.be.equal(bigInt(params1.newState).toString());
+
+    // 3. migrate 
+    const { state: stateV3 } = await stateContractMigrationHelper.upgradeContract(stateContractInstance);
+  
+    // 4. publish second state
+    const params2 = await publishState(stateV3, stateTransitionsWithProofs[1]);
+    const res2 = await stateV3.getStateInfoById(params2.id);
+    expect(res2.state).to.be.equal(bigInt(params2.newState).toString());
+  
+    // 5. check _defaultIdType is not initialized
+    await expect(stateV3.getDefaultIdType()).to.be.revertedWith(
+      "Default Id Type is not initialized"
+    );
+    await expect(stateV3.transitStateGeneric(
+        stateTransitionsWithNoProofs[0].id, 
+        stateTransitionsWithNoProofs[0].oldState, 
+        stateTransitionsWithNoProofs[0].newState,
+        stateTransitionsWithNoProofs[0].isOldStateGenesis,
+        1,
+        []
+      )).to.be.revertedWith(
+      "Default Id Type is not initialized"
+    );
```

Comment:
The tests here should kind of a smoke tests. I don't think there is any reason to check anything except a single positive case.
The rest negative and positive tests should already exist at this smart contract regular unit tests.